sdk/data/azcosmos/cosmos_container.go

```diff
@@ -358,6 +359,226 @@ func (c *ContainerClient) ReadItem(
 	return newItemResponse(azResponse)
 }
 
+// PatchItem patches an item in a Cosmos container.
+// ctx - The context for the request.
+// partitionKey - The partition key for the item.
+// itemId - The id of the item to patch.
+// item - The content to be used to patch.
+// o - Options for the operation.
+func (c *ContainerClient) PatchItem(
+	ctx context.Context,
+	partitionKey PartitionKey,
+	itemId string,
+	item []byte,
+	o *ItemOptions) (ItemResponse, error) {
+	h := headerOptionsOverride{
+		partitionKey: &partitionKey,
+	}
+
+	if o == nil {
+		o = &ItemOptions{}
+	}
+
+	operationContext := pipelineRequestOptions{
+		resourceType:          resourceTypeDocument,
+		resourceAddress:       createLink(c.link, pathSegmentDocument, itemId),
+		headerOptionsOverride: &h}
+
+	path, err := generatePathForNameBased(resourceTypeDocument, operationContext.resourceAddress, false)
+	if err != nil {
+		return ItemResponse{}, err
+	}
+
+	azResponse, err := c.database.client.sendPatchRequest(
+		path,
+		ctx,
+		item,
+		operationContext,
+		o,
+		nil)
+	if err != nil {
+		return ItemResponse{}, err
+	}
+
+	return newItemResponse(azResponse)
+}
+
+// PatchIncrementItem increments a value at the target path within an item in a Cosmos container.
+// ctx - The context for the request.
+// partitionKey - The partition key for the item.
+// itemId - The id of the item to patch.
+// path - The path within the item.
+// value - The value which will be incremented.
+// o - Options for the operation.
+func (c *ContainerClient) PatchIncrementItem(
+	ctx context.Context,
+	partitionKey PartitionKey,
+	itemId string,
+	path string,
+	value interface{},
+	o *ItemOptions) (ItemResponse, error) {
+
+	patchItem := map[string]any{
+		"operations": []struct {
+			Op    string      `json:"op"`
+			Path  string      `json:"path"`
+			Value interface{} `json:"value"`
+		}{
+			{
+				Op:    "increment",
+				Path:  path,
+				Value: value,
+			},
+		},
+	}
+
+	marshalledPatch, err := json.Marshal(patchItem)
+	if err != nil {
+		return ItemResponse{}, err
+	}
+	return c.PatchItem(ctx, partitionKey, itemId, marshalledPatch, o)
+}
+
+// PatchSetItem sets a value at the target path within an item in a Cosmos container.
+// ctx - The context for the request.
+// partitionKey - The partition key for the item.
+// itemId - The id of the item to patch.
+// path - The path within the item.
+// value - The value which will be incremented.
+// o - Options for the operation.
+func (c *ContainerClient) PatchSetItem(
+	ctx context.Context,
+	partitionKey PartitionKey,
+	itemId string,
+	path string,
+	value interface{},
```

Comment:
Having a single method that receives a list of Patch Operations would reduce the need for all these other methods

Reply:
Will keep working on this